### ISSUE TITLE
aws-s3: Unable to use Bucket.fromBucketName on legacy s3 bucket with underscore in the name

### ISSUE THREAD

**Issue by NoahCarnahan** [aws/aws-cdk#22640]
### Describe the bug

I'm attempting to reference an s3 bucket in my cdk code that was created years ago outside of cdk. The bucket has an underscore in its name, which was legal for buckets before March 1, 2018 (new buckets can't have underscores in their names). I've tried referencing the bucket with both `Bucket.fromBucketName` and `Bucket.fromBucketAttributes`, but both throw this error:
> jsii.errors.JSIIError: Invalid S3 bucket name (value: xxxx_xxxxxx)
> Bucket name must only contain lowercase characters and the symbols, period (.) and dash (-) (offset: 4)

(note that I've censored the actual bucket name for privacy)

Also note that I'm using the python version of the cdk library. Is there any way I can reference this bucket from CDK?

### Expected Behavior

I expect to be able to import the bucket.

### Current Behavior

Error message is:
```
jsii.errors.JSIIError: Invalid S3 bucket name (value: xxxx_xxxxxx)
Bucket name must only contain lowercase characters and the symbols, period (.) and dash (-) (offset: 4)
```

Stack trace (paths modified for privacy)
```
  File "<project root>/.venv/lib/python3.9/site-packages/aws_cdk/aws_s3/__init__.py", line 16101, in from_bucket_attributes
    return typing.cast(IBucket, jsii.sinvoke(cls, "fromBucketAttributes", [scope, id, attrs]))
  File "<project root>/.venv/lib/python3.9/site-packages/jsii/_kernel/__init__.py", line 148, in wrapped
    return _recursize_dereference(kernel, fn(kernel, *args, **kwargs))
  File "<project root>/.venv/lib/python3.9/site-packages/jsii/_kernel/__init__.py", line 405, in sinvoke
    response = self.provider.sinvoke(
  File "<project root>/.venv/lib/python3.9/site-packages/jsii/_kernel/providers/process.py", line 365, in sinvoke
    return self._process.send(request, InvokeResponse)
  File "<project root>/.venv/lib/python3.9/site-packages/jsii/_kernel/providers/process.py", line 329, in send
    raise JSIIError(resp.error) from JavaScriptError(resp.stack)
jsii.errors.JSIIError: Invalid S3 bucket name (value: xxxx_xxxxxx)
Bucket name must only contain lowercase characters and the symbols, period (.) and dash (-) (offset: 4)
```

### Reproduction Steps

```python
from aws_cdk import aws_s3 as s3

bucket = s3.Bucket.from_bucket_attributes(
    self,
    "foo-bar",
    account="myAccount",
    bucket_arn="arn:aws:s3:::foo_bar"
)
```

### CDK CLI Version

2.46.0 (build 5a0595e)

### Node.js Version

v18.0.0

### OS

macOS 12.6

### Language

Python

### Language Version

Python 3.9.12

**Comment by peterwoodworth** [comment]
We do not currently support this, thanks for calling this out @NoahCarnahan,

This code is checking for only letters, numbers, periods, and hyphens, and throws an error if any other character is found: https://github.com/aws/aws-cdk/blob/3528e3de5b5dd0a520624d4af73413370dcdc434/packages/%40aws-cdk/aws-s3/lib/bucket.ts#L1727-L1729

We shouldn't be throwing an error for underscores for cases like yours, so we should be checking for underscores as well here.

Until we fix this, there isn't a great workaround that I can think of. Off the top of my head, you could supply an arn without the underscore, and then use the bucket normally. But make sure to use escape hatches to override the arn/name wherever your bucket arn or name is passed in (you can view the template to see where your imported bucket is being used exactly by seeing where its arn is in the template)

**Comment by chicken-teriyaki-cup-rice** [comment]
I'd like to fix this.

**Comment by chicken-teriyaki-cup-rice** [comment]
Made a fix with #22797. 

Ran integration tests. 
Expected result: passed
Actual result: passed


<img width="700" alt="Screenshot 2022-11-06 at 00 41 32" src="https://user-images.githubusercontent.com/91502105/200154653-4e17c006-d474-4650-98c4-d39806fcbfd6.png">

**Comment by NoahCarnahan** [comment]
Thanks both for looking into this
